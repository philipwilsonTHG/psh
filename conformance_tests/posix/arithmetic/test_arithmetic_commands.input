# Test arithmetic commands (standalone ((...)) syntax)

# Test basic arithmetic command with exit status
((2 + 2))
echo "Exit status of ((2 + 2)): $?"

((0))
echo "Exit status of ((0)): $?"

((5))
echo "Exit status of ((5)): $?"

# Test arithmetic command in conditionals
if ((10 > 5)); then
    echo "10 is greater than 5"
fi

if ((0)); then
    echo "This should not print"
else
    echo "Zero is false in arithmetic"
fi

# Test arithmetic assignment in commands
((result = 10 * 3))
echo "Result: $result"

# Test increment in arithmetic commands
((result++))
echo "After increment: $result"

# Test complex expression
((final = result * 2 + 5))
echo "Final: $final"