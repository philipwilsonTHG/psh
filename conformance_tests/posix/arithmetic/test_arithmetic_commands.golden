Exit status of ((2 + 2)): 0
Exit status of ((0)): 1
Exit status of ((5)): 0
10 is greater than 5
Zero is false in arithmetic
Result: 30
After increment: 31
Final: 67
